src/core/qgsvectorlayer.cpp

```diff
@@ -2485,13 +2519,15 @@ bool QgsVectorLayer::writeSld( QDomNode &node, QDomDocument &doc, QString &error
 }
 
 
-bool QgsVectorLayer::changeGeometry( QgsFeatureId fid, const QgsGeometry &geom, bool skipDefaultValue )
+bool QgsVectorLayer::changeGeometry( QgsFeatureId fid, QgsGeometry &geom, bool skipDefaultValue )
 {
   if ( !mEditBuffer || !mDataProvider )
   {
     return false;
   }
 
+  applyGeometryFixes( geom );
```

Comment:
Seeing other bits of code, it feels this should be also guarded by if (geometryFixesEnabled())